### PR TITLE
Add Mock Redis

### PR DESCRIPTION
**Pull request by s3131212**

### CODE REVIEW

File: common/rpc-directory/mock-redis.mjs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    get(key, cb) {
+        if (this._data.has(key)){
+            cb(null, this._data.get(key));
+        } else {
+            cb(null, null); // If the key does not exist the special value nil is returned.
+        }
+    }
+
+    set(key, value, cb) {
```
Comment (john0312) [inline common/rpc-directory/mock-redis.mjs:37]: We currently use the form set([key, value], cb).

Comment (john0312) [reply common/rpc-directory/mock-redis.mjs:37]: and we need to support 'NX' attribute.

File: common/rpc-directory/mock-redis.mjs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    on(event, callback) {
+        if (event == 'message') {
+            this._mockRedis.callbacks.push(callback);
+        }
+    }
+
+    hget(key, cb) {
+        if (this._data.has(key)){
+            cb(null, this._data.get(key));
```
Comment (john0312) [inline common/rpc-directory/mock-redis.mjs:64]: hget/hset works a level deeper.
i.e. cb(null, this._data.get(key).get(field))

**Review by john0312 — APPROVED** [review]

File: common/rpc-directory/mock-redis.mjs (hunk 1 above)
Comment (s3131212) [reply common/rpc-directory/mock-redis.mjs]: Resolved.

File: common/rpc-directory/mock-redis.mjs (hunk 2 above)
Comment (s3131212) [reply common/rpc-directory/mock-redis.mjs]: Resolved. I also changed the signature of `hget` and `hset`, please check if those are correct.

**Comment by john0312** [comment]
Still LGTM.

**Pull request merged**